lib/Extension/LanguageServerEvaluatableExpression/Handler/EvaluatableExpressionHandler.php

```diff
@@ -0,0 +1,106 @@
+use Phpactor\LanguageServerProtocol\Position;
+use Phpactor\LanguageServerProtocol\ServerCapabilities;
+use Phpactor\LanguageServerProtocol\TextDocumentIdentifier;
+use Phpactor\LanguageServer\Core\Handler\CanRegisterCapabilities;
+use Phpactor\LanguageServer\Core\Handler\Handler;
+use Phpactor\LanguageServer\Core\Workspace\Workspace;
+use Phpactor\TextDocument\TextDocumentBuilder;
+use Phpactor\WorseReflection\Core\Util\NodeUtil;
+use Phpactor\TextDocument\ByteOffsetRange;
+use Phpactor\LanguageServerProtocol\Range;
+
+class EvaluatableExpressionHandler implements Handler, CanRegisterCapabilities
+{
+    public function __construct(
+        private Workspace $workspace,
+        private Parser $parser,
+    ) {
+    }
+
+    public function methods(): array
+    {
+        return [
+            'textDocument/xevaluatableExpression' => 'xevaluatableExpression',
+        ];
+    }
+
+    /**
+     * @return Promise<EvaluatableExpression|null>
+     */
+    public function xevaluatableExpression(
+        TextDocumentIdentifier $textDocument,
+        Position $position
+    ): Promise {
+        return call(function () use ($textDocument, $position) {
+            $document = $this->workspace->get($textDocument->uri);
+            $offset = PositionConverter::positionToByteOffset($position, $document->text);
+            $document = TextDocumentBuilder::create($document->text)
+                ->uri($document->uri)
+                ->language('php')
+                ->build();
+
+            $char = substr($document, $offset->toInt(), 1);
+
+            // do not provide evaluatable for whitespace
+            if (trim($char) == '') {
+                return null;
+            }
+
+            $rootNode = $this->parser->parseSourceFile((string) $document, $document->uri()?->__toString());
+            $node = $rootNode->getDescendantNodeAtPosition($offset->toInt());
+            $ret = $this->nodeToEvaluatable($node);
+            return $ret;
+        });
+    }
+
+    public function registerCapabiltiies(ServerCapabilities $capabilities): void
+    {
+        $capabilities->experimental['xevaluatableExpressionProvider'] = true;
+    }
+
+    private function nodeToEvaluatable(Node $node): ?EvaluatableExpression
+    {
+        if ($node instanceof Node\Parameter) {
+            return
+                new EvaluatableExpression(
+                    expression: $node->variableName->getText($node->getFileContents()),
+                    range: $this->byteOffsetRangeForNode($node->variableName, $node),
+                );
+        }
+        if (
+            $node instanceof Node\Expression\Variable ||
+            $node instanceof Node\Expression\SubscriptExpression ||
+            $node instanceof Node\Expression\MemberAccessExpression ||
+            (($node2 = $node->getFirstAncestor(Node\Expression\SubscriptExpression::class)) && ($node = $node2))
+        ) {
```

Comment:
maybe extract `$this->isXXXExpression` and avoid assigning in a conditional expression?

Reply:
This was a style mostly used in php-intellisense. I'll split up the code and make it more readable.

Reply:
Reduced the code a bit... I tried to be smart with `match` but it's not allowed to do an assignment inside a match case...